### PR TITLE
Add Specter, Multifacet-Recommender and the full Ensemble

### PR DESCRIPTION
**Pull request by ameyagodbole**
- Add model `specter` ([base repository](https://github.com/allenai/specter)) under `expertise/models/multifacet_recommender/`
- Add model `mfr` (multifacet_recommender) under `expertise/models/multifacet_recommender/`
- Add model `specter+mfr` ensemble under `expertise/models/multifacet_recommender/`. This is a linear combination of the `specter` and `mfr` scores with a parameter `merge_alpha`

### CODE REVIEW

**Comment by ameyagodbole** [comment]
TODO:
[ ] Add tests and configure CircleCI
[ ] Remove redundant code under `expertise/models/multifacet_recommender/mfr_src/` (low priority)

File: README.md [hunk 1]
```diff
@@ -33,6 +33,10 @@ conda install faiss-cpu -c pytorch
 ```
 [Here](https://github.com/facebookresearch/faiss/blob/master/INSTALL.md) you can find the above installation command.
 
+If you plan to use SPECTER, follow the set-up steps 1 and 2 described [here](https://github.com/allenai/specter#how-to-use-the-pretrained-model). Provide the SPECTER repository path in the config as `model_params.specter_dir`.
+
+If you plan to use Multifacet-Recommender, download the checkpoint files from [here](https://drive.google.com/file/d/1_mWkQ1dr_Vl121WZkbNyNMV3G_bmoQ6s/view?usp=sharing), extract it, and pass the path to `feature_vocab_file` and `model_checkpoint_dir`
```
Comment (melisabok) [inline README.md:38]: how do I download this using `wget`?

Comment (melisabok) [reply README.md:38]: I'm assuming that `model_checkpoint_dir` is `multifacet_recommender_data/mfr_model_checkpoint` in the compressed file, is this correct?

File: README.md [hunk 2]
```diff
@@ -255,6 +259,110 @@ Here is an example:
 }
 ```
 
+#### SPECTER specific parameters (affinity scores):
```
Comment (melisabok) [inline README.md:262]: Should we assume these three new models need to be run using the virtulenv `specter` specified here: https://github.com/allenai/specter#how-to-use-the-pretrained-model?

File: README.md [hunk 3]
```diff
@@ -255,6 +259,110 @@ Here is an example:
...
+    },
+    "model": "specter",
+    "model_params": {
+        "specter_dir": "../specter/",
+        "work_dir": "./",
+        "average_score": false,
+        "max_score": true,
+        "use_cuda": true,
+        "batch_size": 16,
+        "publications_path": "./",
```
Comment (melisabok) [inline README.md:288]: can we use `dataset` property and remove these two: `publications_path` and `submissions_path`?

File: README.md [hunk 4]
```diff
@@ -255,6 +259,110 @@ Here is an example:
...
+    "dataset": {
+        "directory": "./"
+    },
+    "model": "specter+mfr",
+    "model_params": {
+        "specter_dir": "../specter/",
+        "average_score": false,
+        "max_score": true,
+        "specter_batch_size": 16,
+        "publications_path": "./",
```
Comment (melisabok) [inline README.md:352]: same here

File: README.md (hunk 3 above)
Comment (melisabok) [reply README.md:288]: never mind, this seems to be something else, I'm not sure why we need different paths.

File: README.md [hunk 5]
```diff
@@ -255,6 +259,110 @@ Here is an example:
...
+        "work_dir": "./",
+        "epochs": 100,
+        "batch_size": 50,
+        "use_cuda": true,
+        "scores_path": "./"
+    }
+}
+```
+
+#### SPECTER+MFR ensemble specific parameters (affinity scores):
```
Comment (melisabok) [inline README.md:323]: I'm testing specter+mfr and I'm getting this error:

```
3316batches [09:59,  5.53batches/s]
MFR:
Featurizing publications...
Traceback (most recent call last):
  File "/home/mbok/miniconda3/envs/specter/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/mbok/miniconda3/envs/specter/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/mbok/openreview-expertise/expertise/run.py", line 145, in <module>
    mfr_publications_path=None, skip_specter=config['model_params'].get('skip_specter', False))
  File "/home/mbok/openreview-expertise/expertise/models/multifacet_recommender/ensemble.py", line 58, in embed_publications
    self.mfr_predictor.embed_publications(mfr_publications_path)
  File "/home/mbok/openreview-expertise/expertise/models/multifacet_recommender/multifacet_recommender.py", line 610, in embed_publications
    abstracts_corpus_tokens = self._tokenize_text(abstracts_corpus)
  File "/home/mbok/openreview-expertise/expertise/models/multifacet_recommender/multifacet_recommender.py", line 297, in _tokenize_text
    w_list.append([w.text for w in nlp.tokenizer(line)])
  File "tokenizer.pyx", line 85, in spacy.tokenizer.Tokenizer.__call__
TypeError: object of type 'NoneType' has no len()

If you suspect this is an IPython 7.18.1 bug, please report it at:
    https://github.com/ipython/ipython/issues
or send an email to the mailing list at ipython-dev@python.org

You can print a more detailed traceback right now with "%tb", or use "%debug"
to interactively debug it.

Extra-detailed tracebacks for bug-reporting purposes can be enabled via:
    %config Application.verbose_crash=True

09/24/2020 10:04:11 - INFO - allennlp.models.archival -   removing temporary unarchived model dir at /tmp/tmpcdzpjyl2
```

Comment (ameyagodbole) [reply README.md:323]: Pushed some changes. Can you try again?

File: README.md (hunk 2 above)
Comment (ameyagodbole) [reply README.md:262]: I'll add the steps explicitly to the README since we want it installed in the same environment as openreview-expertize

**Review by melisabok — APPROVED** [review]

**Pull request merged**